src/python/pants/source/source_root.py

```diff
@@ -346,6 +346,8 @@ def fixed(self):
 
   def _do_add_pattern(self, pattern, langs, category):
     keys = pattern.split(os.path.sep)
+    if keys[-1] == '':
+      keys = keys[:-1]
```

Comment:
Could you add a test for this? Also, does this apply just for paths like `x/` and `^/`, or should we also normalize `x//y`?